### ISSUE TITLE
WebGPURenderer: errors thrown rendering line segments

### ISSUE THREAD

**Issue by WestLangley** [mrdoob/three.js#28927]
### Description

I get the following error if `AxesHelper` is rendered prior to `VertexNormalsHelper` (WebGPU backend only):

<img width="731" alt="Screenshot 2024-07-20 at 8 05 05 PM" src="https://github.com/user-attachments/assets/c1f725de-f0a6-44ff-b8bf-8ffedd002719">

### Reproduction steps

See below.

### Code

In `webgpu_materials_matcap.html`, add these lines in the loader call-back:

```js
// Switch the order of these two lines and the scene renders correctly.

scene.add( new THREE.AxesHelper( 5 ) );

scene.add( new VertexNormalsHelper( mesh, 0.1, 0xff0000 ) );
```

Also include
```js
import { VertexNormalsHelper } from 'three/addons/helpers/VertexNormalsHelper.js';
```

### Live example

https://jsfiddle.net/62tzex3v/3/

### Version

r167 dev

### Device

Desktop

### Browser

Chrome

### OS

MacOS

**Comment by cmhhelgeson** [comment]
TLDR: Did some testing and VertexNormalsHelper is rendering using AxesHelper pipeline and vertex state.

Maybe somebody's already found this, but here were the modifications I used to test the code.

1. Only add the helpers to the scene without the mesh.
2. Add a suffix to the renderPipeline name with the number of vertex buffers it sets during a draw call. 

Axes Helper uses two vertex buffers, one for positions and one for color. It's pipeline is named renderPipeline_2.

VertexNormalsHelper uses just one vertex buffer for positions. It's pipeline is named renderPipeline_1.

When AxesHelper is added to a scene before VertexNormalsHelper, it's pipeline is run beforehand. So renderPipeline_2 first then renderPipeline_1.

3. At the end of each WebGPUBackend.draw call, add this code to unset the vertex buffers.

```
for ( let i = 0; i < vertexBuffers.length; i ++ ) {

     passEncoderGPU.setVertexBuffer( i, null );

}
		
```
And voila, the inciting problem, the render pass being used to render the VertexNormalsHelper is still using the Axes Helper render pipeline, since it expects vertex buffer slot 1 to be set. 

<img width="1106" alt="image" src="https://github.com/user-attachments/assets/cbf8a955-38c0-4024-a954-d1f4aee001a0">

**Comment by Mugen87** [comment]
For explanation: An error in a cache key computation produces a wrong cache key. That means the wrong node builder was selected returning the wrong shaders for the render object.
